clang/lib/Parse/ParseHLSL.cpp

```diff
@@ -141,6 +141,8 @@ void Parser::ParseHLSLAnnotations(ParsedAttributes &Attrs,
     return;
   }
 
+  II = PP.getIdentifierInfo(II->getName().lower());
```

Comment:
Is this used?

Reply:
Yes, II is passed two places.

Reply:
Can we keep the original identifier around and use that for the diagnostics so that the diagnostic text matches the capitalization in the source?

I often find myself copy and pasting error message text to then grep codebases, so it would be nice if we preserved the spelling.

Reply:
Okay that might requirement a significant change. I'll look into it